### PR TITLE
Update test according to CLDR38

### PR DESCRIPTION
**Pull request by FrankYFTang**
@anba

### CODE REVIEW

File: test/intl402/Intl/getCanonicalLocales/preferred-grandfathered.js [hunk 1]
```diff
@@ -62,7 +62,7 @@ var regularGrandfatheredNonUTS35 = [
 
 var regularGrandfatheredUTS35 = {
   "art-lojban": "jbo",
-  "cel-gaulish": "xtg-x-cel-gaulish",
+  "cel-gaulish": "xtg",
```
Comment (rwaldron) [inline test/intl402/Intl/getCanonicalLocales/preferred-grandfathered.js:65]: Should this also be updated in `harness/testIntl.js`?

**Review by rwaldron — CHANGES_REQUESTED** [review]
I just regenerated all of the mappings in `harness/testIntl.js` using `./tools/misc/make_intl_data.py --version 38 --out generated.js` and there are substantial updates there, you can cherry pick that commit from here: https://github.com/tc39/test262/tree/cldr-38

File: test/intl402/Intl/getCanonicalLocales/preferred-grandfathered.js (hunk 1 above)
Comment (FrankYFTang) [reply test/intl402/Intl/getCanonicalLocales/preferred-grandfathered.js:65]: sure, I am not aware of such tool

**Comment by FrankYFTang** [comment]
@ryzokuken

**Comment by FrankYFTang** [comment]
ping

**Comment by FrankYFTang** [comment]
@zbraniecki @anba please help to review

**Comment by FrankYFTang** [comment]
ping

**Comment by FrankYFTang** [comment]
@mathiasbynens

**Review by zbraniecki — APPROVED** [review]
lgtm

**Review by rwaldron — APPROVED** [review]

**Pull request merged**